app/src/main/java/app/insti/fragment/FileComplaintFragment.java

```diff
@@ -0,0 +1,789 @@
+import app.insti.activity.MainActivity;
+import app.insti.adapter.ImageViewPagerAdapter;
+import app.insti.api.LocationAPIUtils;
+import app.insti.api.RetrofitInterface;
+import app.insti.api.request.ComplaintCreateRequest;
+import app.insti.api.request.ImageUploadRequest;
+import app.insti.api.response.ComplaintCreateResponse;
+import app.insti.api.response.ImageUploadResponse;
+import app.insti.utils.TagCategories;
+import me.relex.circleindicator.CircleIndicator;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+import static app.insti.Constants.MY_PERMISSIONS_REQUEST_LOCATION;
+import static app.insti.Constants.MY_PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE;
+import static app.insti.Constants.RESULT_LOAD_IMAGE;
+
+public class FileComplaintFragment extends Fragment {
+
+    private static final String TAG = FileComplaintFragment.class.getSimpleName();
+    private static FileComplaintFragment mainactivity;
+    private Button buttonSubmit;
+    private FloatingActionButton floatingActionButton;
+    private CustomAutoCompleteTextView autoCompleteTextView;
+    private EditText editTextSuggestions;
+    private EditText editTextTags;
+    private MapView mMapView;
+    GoogleMap googleMap;
+    private TagView tagView;
+    private TagView tagViewPopulate;
+    private ScrollView tagsLayout;
+    private LatLng Location;
+    private String Name;
+    private String Address;
+    private List<String> Tags;
+    private ArrayList<TagClass> tagList;
+    private List<String> uploadedImagesUrl = new ArrayList<>();
+    private int cursor = 1;
+    private List<TagClass> tagList2 = new ArrayList<>();
+
+    private String base64Image;
+    private ImageViewPagerAdapter imageViewPagerAdapter;
+    private ViewPager viewPager;
+    private CircleIndicator indicator;
+    private Button buttonAnalysis;
+    private LinearLayout layout_buttons;
+    String userId;
+    View view;
+    NestedScrollView nestedScrollView;
+    LinearLayout linearLayoutAddImage;
+    private boolean GPSIsSetup = false;
+    FusedLocationProviderClient mFusedLocationClient;
+    ProgressDialog progressDialog;
+
+    public FileComplaintFragment() {
+        // Required empty public constructor
+    }
+
+    public static FileComplaintFragment getMainActivity() {
+        return mainactivity;
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        android.app.FragmentManager fragmentManager = getActivity().getFragmentManager();
+        PlaceAutocompleteFragment fragment = (PlaceAutocompleteFragment) fragmentManager.findFragmentById(R.id.place_autocomplete_fragment);
+        android.app.FragmentTransaction ft = fragmentManager.beginTransaction();
+        ft.remove(fragment);
+        ft.commit();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        Toast.makeText(getContext(), getString(R.string.initial_message_file_complaint), Toast.LENGTH_SHORT).show();
```

Comment:
Don't show this here. It is
1) Detected as a screen overlay, so the user cannot allow on MM+
2) Plainly unnecessary

Reply:
This was told by ma'am in today's meeting to add a toast on start of this fragment

Reply:
Granted Permission for what?

Reply:
Map

Reply:
Ohk ok. Got it 👍